go/plugins/dotprompt/picoschema_test.go

```diff
@@ -15,158 +15,87 @@
-			        "required": ["req"]
-			}`,
-		},
-		{
-			description: "array of scalars, with and without description",
-			yaml: `schema:
-  tags(array, list of tags): string, the tag
-  vector(array): number`,
-			want: `{
-			        "type": "object",
-			        "properties": {
-			          "tags": {
-			            "type": "array",
-			            "description": "list of tags",
-			            "items": { "type": "string", "description": "the tag" }
-			          },
-			          "vector": { "type": "array", "items": { "type": "number" } }
-			        },
-			        "required": ["tags", "vector"]
-			}`,
-		},
-		{
-			description: "nested object in array and out",
-			yaml: `schema:
-  obj?(object, a nested object):
-    nest1?: string
-  arr(array, array of objects):
-    nest2?: boolean`,
-			want: `{
-			        "type": "object",
-			        "properties": {
-			          "obj": {
-			            "type": "object",
-			            "description": "a nested object",
-			            "properties": { "nest1": { "type": "string" } }
-			          },
-			          "arr": {
-			            "type": "array",
-			            "description": "array of objects",
-			            "items": {
-			              "type": "object",
-			              "properties": { "nest2": { "type": "boolean" } }
-			            }
-			          }
-			        },
-			        "required": ["arr"]
-			}`,
-		},
-		{
-			description: "simple json schema type",
-			yaml: `schema:
-  type: string`,
-			want: `{ "type": "string" }`,
-		},
-		{
-			description: "simple json schema type",
-			yaml: `schema:
-  type: string`,
-			want: `{ "type": "string" }`,
-		},
-		{
-			description: "inferred json schema from properties",
-			yaml: `schema:
-  properties:
-    foo: {type: string}`,
-			want: `{ "type": "object", "properties": { "foo": { "type": "string" } } }`,
-		},
+	type test struct {
+		Description string
+		YAML        string
+		Want        map[string]any
+	}
+
+	convertSchema(&jsonschema.Schema{})
+	data, err := os.ReadFile(filepath.FromSlash("../../../js/plugins/dotprompt/tests/picoschema_tests.yaml"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var tests []test
+	if err := yaml.Unmarshal(data, &tests); err != nil {
+		t.Fatal(err)
+	}
+
+	skip := map[string]bool{
+		"required field":                 true,
+		"nested object in array and out": true,
 	}
 
 	for _, test := range tests {
-		var val any
-		if err := yaml.Unmarshal([]byte(test.yaml), &val); err != nil {
-			t.Errorf("%s YAML unmarshal failure: %v", test.description, err)
-			continue
-		}
+		t.Run(test.Description, func(t *testing.T) {
+			if skip[test.Description] {
+				t.Skip("no support for type as an array")
+			}
+			var val any
+			if err := yaml.Unmarshal([]byte(test.YAML), &val); err != nil {
+				t.Fatalf("YAML unmarshal failure: %v", err)
+			}
 
-		// The tests, copied from TypeScript, use a schema field.
-		val = val.(map[string]any)["schema"]
+			// The tests, copied from TypeScript, use a schema field.
```

Comment:
You can drop "copied from TypeScript" now.

Reply:
Done.